Add React to a Website (#17)
reactjs/pl.reactjs.org: content/docs/add-react-to-a-website.md

---
id: add-react-to-a-website
title: Dodaj Reacta do swojej strony
permalink: docs/add-react-to-a-website.html
redirect_from:
  - "docs/add-react-to-an-existing-app.html"
prev: getting-started.html
next: create-a-new-react-app.html
---

> Używaj Reacta wybiórczo w miarę potrzeb.

Reviewer: Proszę, wywal to zdanie :D Tak mnie razi, że za chwilę pojawia się jego kopia, że nie mogę na to patrzeć :P

Author: > Proszę, wywal to zdanie :D Tak mnie razi to, że za chwilę pojawia się jego kopia, że nie mogę na to patrzeć :P

Ale wtedy cały nagłówek i formatowanie się rozwali - masz jakieś propozycje jak tego uniknać?

Author: Ale wtedy cały nagłówek i formatowanie się rozwali - masz jakieś propozycje jak tego uniknać?

> Użyj tak mało lub tak dużo Reacta ile potrzebujesz.

Reviewer: ```suggestion
Używaj Reacta wybiórczo w zależności od potrzeb.
```

> Użyj tak mało lub tak dużo Reacta ile potrzebujesz.

Reviewer: Ja bym ten akapit całkowicie usunął. Nie wnosi on niczego nowego, a poza tym podobny fragment znajduje się w następnym akapicie...

Author: OK - wszystko poprawione poza tym akapitem? Sugerujesz zeby usunac tylko: Użyj tak mało lub tak dużo Reacta ile potrzebujesz.

Nie lepiej zostwic zeby bylo podobnie jak w wersji angielskiej?

Author: Dla mnie +1

> Biblioteka React została zaprojektowana z myślą o stopniowym wdrażaniu. Dzięki temu zawarte w niej rozwiązania możesz stosować wybiórczo w zależności od potrzeb. Być może, jedyne co chcesz to dodać trochę interaktywności do istniejącej strony. Komponenty React są świetne do zrobienia tego.

Reviewer: ```suggestion
Biblioteka React została zaprojektowana z myślą o stopniowym wdrażaniu. Dzięki temu zawarte w niej rozwiązania możesz stosować wybiórczo w zależności od potrzeb. Być może chcesz jedynie dodać trochę interaktywności do istniejącej strony. Komponenty reactowe świetnie się do tego nadają.
```

> Większość stron internetowych nie jest i nie musi być aplikacjami jednostronowymi. Z **kilkoma liniami kodu i bez narzędzi kompilacji**, wypróbuj React w małej części Twojej strony. Następnie możesz stopniowo zwiększać jego wykorzystanie lub przechowywać w kilku dynamicznych widżetach.

Reviewer: ```suggestion
Większość stron internetowych nie jest i nie musi być aplikacjami jednostronowymi (ang. *single-page apps*). Wypróbuj Reacta w niewielkiej części swojej strony za pomocą **kilku linii kodu i bez narzędzi kompilacji**. Następnie możesz stopniowo zwiększać jego wykorzystanie lub zostać przy kilku dynamicznych widżetach.
```

---

> - [Dodaj React w minutę](#add-react-in-one-minute)

Reviewer: ```suggestion
- [Dodaj Reacta w minutę](#add-react-in-one-minute)
```

> - [Dodaj Reacta w minutę](#add-react-in-one-minute)

Reviewer 2: Wydaje mi się, że w polskiej mowie nie posługujemy się raczej wyrażeniem "zrobić coś w minutę".  Częściej zdaje się, mówimy "zrobić coś w dwie/pięć minut". Proponowałbym jednak, trzymać się utartych związków frazeologicznych np. coś w staroświeckim stylu "mgnienie oka" 

```suggestion
- [Dodaj Reacta w mgnieniu oka](#add-react-in-one-minute)
```
lub wręcz:

```suggestion
- [Błyskawiczne dodawanie Reacta](#add-react-in-one-minute)
```

> - [Opcjonalnie: Wypróbuj React z JSX](#optional-try-react-with-jsx) (bundler nie jest wymagany!)

Reviewer: ```suggestion
- [Opcjonalnie: Wypróbuj Reacta z JSX](#optional-try-react-with-jsx) (bundler nie jest wymagany!)
```

> ## Dodaj React w minutę {#add-react-in-one-minute}

Reviewer: ```suggestion
## Dodaj Reacta w minutę {#add-react-in-one-minute}
```

> ## Dodaj Reacta w minutę {#add-react-in-one-minute}

Reviewer 2: jw.

Author: Dodane: w mgnieniu oka

> W tej sekcji pokażemy jak dodać komponent Reakta do strony internetowej. Możesz to zrobić korzystając z własnej strony internetowej lub stwórz nową pustą stronę.

Reviewer: ```suggestion
W tej sekcji pokażemy, jak dodać komponent reactowy do strony internetowej. Możesz to zrobić korzystając z własnej strony internetowej lub stworzyć nowy plik HTML.
```

> Nie będziemy korzystali z żadnych skomplikowanych narzędzi lub wymagań instalacyjnych -- **aby ukończyć tę sekcję, jedyne czego potrzebujesz to połączenie internetowe i minuta wolnego czasu.**

Reviewer: ```suggestion
Nie będziemy instalowali żadnych zależności ani korzystali ze skomplikowanych narzędzi -- **aby ukończyć ten rozdział, będziesz potrzebować jedynie połączenia internetowego i minuty wolnego czasu.**
```

> Nie będziemy instalowali żadnych zależności ani korzystali ze skomplikowanych narzędzi -- **aby ukończyć ten rozdział, będziesz potrzebować jedynie połączenia internetowego i minuty wolnego czasu.**

Reviewer 2: ```suggestion
Nie będziemy instalowali żadnych zależności ani korzystali ze skomplikowanych narzędzi -- **aby ukończyć ten rozdział, będziesz potrzebować jedynie połączenia internetowego i odrobiny wolnego czasu.**
```

> Opcjonalnie: [Pobierz cały przykład (2KB zipped)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)

Reviewer: ```suggestion
Opcjonalnie: [Pobierz cały przykład (2KB po kompresji)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)
```

### Krok 1: Dodaj kontener do HTML {#step-1-add-a-dom-container-to-the-html}

> Najpierw otwórz stronę internetowa, która chcesz edytować. Dodaj pusty znacznik '<div>' aby zaznaczyć miejsce w którym chcesz wyświetlić coś za pomocą przycisku React. Na przykład:

Reviewer: ```suggestion
Najpierw otwórz stronę internetową, którą chcesz edytować. Dodaj pusty znacznik `<div>`, aby określić miejsce, w którym React będzie wyświetlał treść. Na przykład:
```

> '''html{3}

Reviewer: ```suggestion
```html{3}
```

<!-- ... istniejący kod HTML ... -->

<div id="like_button_container"></div>

<!-- ... istniejący kod HTML ... -->

> '''

Reviewer: ```suggestion
```
```

Reviewer: Zamiast apostrofów używaj `

> Przypisaliśmy temu znacznikowi '<div>' unikalny atrybut 'id'. Pozwoli nam to na późniejsze odnalezienie kodu korzystając z JavaScript oraz na wyświetlenie w nim komponentu React.

Reviewer: ```suggestion
Przypisaliśmy temu znacznikowi `<div>` unikalny atrybut `id`. Pozwoli nam to na późniejsze odnalezienie kodu za pomocą JavaScripta oraz na wyświetlenie w nim komponentu reactowego.
```

>Wskazówka
>

> >Możesz umieścić "kontener" '<div>' **w dowolnym miejscu** wewnątrz znacznika '<body>'. Możesz mieć tyle niezależnych kontenerów na jednej stronie, ile potrzebujesz. Zazwyczaj są one puste -- React zastąpi istniejącą zawartość w kontenerze.

Reviewer: ```suggestion
>Możesz umieścić "kontener" `<div>` **w dowolnym miejscu** wewnątrz znacznika `<body>`. Możesz mieć tyle niezależnych kontenerów na jednej stronie, ile potrzebujesz. Zazwyczaj są one puste -- React i tak zastąpi istniejącą zawartość w kontenerze.
```

> >Możesz umieścić "kontener" `<div>` **w dowolnym miejscu** wewnątrz znacznika `<body>`. Możesz mieć tyle niezależnych kontenerów na jednej stronie, ile potrzebujesz. Zazwyczaj są one puste -- React i tak zastąpi istniejącą zawartość w kontenerze.

Reviewer 2: Proponuję odrobinę mniej literalnie:

```suggestion
>"Kontenery" `<div>` można umieszczać **w dowolnych miejscach** wewnątrz znacznika `<body>`.  Dopuszczlna liczba niezależnych kontenerów na stronie jest nieograniczona.  Zazwyczaj są one puste -- React i tak zastąpi istniejącą zawartość w kontenerze.
```

> ### Krok 2: Dodaj znacznik script {#step-2-add-the-script-tags}

Reviewer: ```suggestion
### Krok 2: Dodaj znaczniki `script` {#step-2-add-the-script-tags}
```

> Następnie, dodaj trzy znaczniki '<script>' do strony internetowej tuż przed zamykającym znacznikiem '</body>':

Reviewer: ```suggestion
Następnie dodaj trzy znaczniki `<script>` do strony HTML tuż przed zamykającym znacznikiem `</body>`:
```

> '''html{5,6,9}

Reviewer: ```suggestion
```html{5,6,9}
```

> <!-- ... pozostały kod HTML ... -->

Reviewer 2: ```suggestion
<!-- ... pozostała część kodu HTML ... -->
```

> <!-- Załaduj React. -->

Reviewer: ```suggestion
<!-- Załaduj Reacta. -->
```

> <!-- Uwaga: podczas wdrażania, zamień "development.js" na "production.min.js". -->

Reviewer: ```suggestion
<!-- Uwaga: podczas wdrożenia na produkcję, zamień "development.js" na "production.min.js". -->
```

> <!-- Uwaga: podczas wdrożenia na produkcję, zamień "development.js" na "production.min.js". -->

Reviewer 2: ```suggestion
<!-- Uwaga: podczas wdrażania aplikacji do środowiska produkcyjnego, zamień "development.js" na "production.min.js". -->
```

<script src="https://unpkg.com/react@16/umd/react.development.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@16/umd/react-dom.development.js" crossorigin></script>

> <!-- Załaduj nasz komponent React. -->

Reviewer: ```suggestion
<!-- Załaduj nasz komponent reactowy. -->
```

<script src="like_button.js"></script>

</body>

> '''

Reviewer: ```suggestion
```
```

Reviewer: Nie zmieniaj "back ticków" na apostrofy, bo strona będzie się błędnie wyświetlać.

Author: To z automatu mi wpadlo, jak sprawdzalem ogonki itp.

> Pierwsze dwa znaczniki wczytują Reacta. Trzeci załaduje kod Twojego komponentu.

Reviewer: ```suggestion
Pierwsze dwa znaczniki wczytują Reacta. Trzeci załaduje kod twojego komponentu.
```

> ### Krok 3: Stwórz komponent Reacta {#step-3-create-a-react-component}

Reviewer: ```suggestion
### Krok 3: Stwórz komponent reactowy {#step-3-create-a-react-component}
```

> Stwórz plik 'like_button.js' w folderze swojej strony.

Reviewer: ```suggestion
Stwórz plik `like_button.js` w folderze, obok pliku HTML ze stroną.
```

> Otwórz **[przykładowy kod](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)** i wklej jego zawartość do pliku, który stworzyłeś.

Reviewer: ```suggestion
Otwórz **[szablon startowy](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)** i wklej jego zawartość do utworzonego pliku.
```

>Wskazówka
>

> >Ten kod definiuje komponent Reakta nazwany 'LikeButton'. Nie przejmuj sie, jeżeli jeszcze tego nie rozumiesz -- wyjaśnimy bloki konstrukcyjne React później w naszym [praktycznym tutorialu](/tutorial/tutorial.html) i [w głównych koncepcjach](/docs/hello-world.html). Na razie pokażmy to na ekranie!

Reviewer: ```suggestion
>Kod ten definiuje komponent reactowy o nazwie `LikeButton`. Nie przejmuj się, jeżeli jeszcze go nie rozumiesz -- podstawy Reacta wyjaśnimy później w naszym [praktycznym samouczku](/tutorial/tutorial.html) i [w rozdziale pt. "Główne idee"](/docs/hello-world.html). Na razie jednak skupmy się na wyświetleniu tego na ekranie!
```

> >Kod ten definiuje komponent reactowy o nazwie `LikeButton`. Nie przejmuj się, jeżeli jeszcze go nie rozumiesz -- podstawy Reacta wyjaśnimy później w naszym [praktycznym samouczku](/tutorial/tutorial.html) i [w rozdziale pt. "Główne idee"](/docs/hello-world.html). Na razie jednak skupmy się na wyświetleniu tego na ekranie!

Reviewer 2: ```suggestion
>Kod ten definiuje komponent reactowy o nazwie `LikeButton`. Nie przejmuj się, jeśli jeszcze czegoś nie rozumiesz na tym etapie -- podstawy Reacta wyjaśnimy później w naszym [praktycznym samouczku](/tutorial/tutorial.html) i [w rozdziale pt. "Główne idee"](/docs/hello-world.html). Na razie jednak skupmy się na wyświetleniu tego na ekranie!
```

> Po **[przykładowym kodzie](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)**, dodaj dwie linijki na na końcu 'like_button.js':

Reviewer: ```suggestion
Pod kodem z **[szablonu startowego](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)** w pliku `like_button.js` dodaj dwie linijki:
```

> '''js{3,4}

Reviewer: ```suggestion
```js{3,4}
```

> // ... przykładowy kod, który wkleiłeś ...

Reviewer: ```suggestion
// ... wklejony kod startowy ...
```

const domContainer = document.querySelector('#like_button_container');
ReactDOM.render(e(LikeButton), domContainer);

> '''

Reviewer: ```suggestion
```
```

> Te dwie linijki kodu znajda znacznik '<div>' który dodaliśmy do naszego kodu HTML w pierwszym kroku i wyświetla nasz przycisk "Like" jako komponent Reacta.

Reviewer: ```suggestion
Te dwie linijki kodu są odpowiedzialne za znalezienie naszego elementu `<div>` z pierwszego kroku i wyświetlenie przycisku `"Like"` pochodzącego z wklejonego komponentu.
```

### To wszystko! {#thats-it}

> Nie ma kroku czwartego. **Właśnie dodałeś pierwszy komponent Reacta do swojej strony.**

Reviewer: ```suggestion
Nie ma kroku czwartego. **Właśnie udało ci się dodać pierwszy komponent reactowy do swojej strony.**
```

> Zapoznaj się z następnymi sekcjami, aby uzyskać więcej podpowiedzi na temat integracji w React.

Reviewer: ```suggestion
Zapoznaj się z następnymi rozdziałami, aby uzyskać więcej informacji na temat integrowania strony z Reactem.
```

**[Zobacz przykładowy kod źródłowy](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605)**

**[Pobierz cały przykład (2KB zipped)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)**

> ### Wskazówka: Wykorzystaj komponent ponownie {#tip-reuse-a-component}

Reviewer: ```suggestion
### Wskazówka: Korzystaj z komponentów wielokrotnie {#tip-reuse-a-component}
```

> ### Wskazówka: Korzystaj z komponentów wielokrotnie {#tip-reuse-a-component}

Reviewer 2: Proponuję:

```suggestion
### Wskazówka:  Wykorzystuj komponenty wielokrotnie {#tip-reuse-a-component}
```

> Często możesz chcieć wyświetlić komponent Reacta w wielu miejscach na stronie. Oto przykład, który trzykrotnie wyświetla przycisk "Like" i przekazuje do niego dane:

Reviewer: ```suggestion
Często zdarza się, że komponent trzeba wyświetlić na stronie kilka razy. Oto przykład, który trzykrotnie wyświetla przycisk "Like" i przekazuje do niego dane:
```

[Zobacz przykładowy kod źródłowy](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda)

> [Pobierz cały przykład (2KB zipped)](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda/archive/9d0dd0ee941fea05fd1357502e5aa348abb84c12.zip)

Reviewer: ```suggestion
[Pobierz cały przykład (2KB po kompresji)](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda/archive/9d0dd0ee941fea05fd1357502e5aa348abb84c12.zip)
```

>Wskazówka
>

> >Ta strategia jest szczegoólnie przydatna gdy części strony oparte na Reackcie sa od siebie odizolowane. Wewnatrz kodu React, łatwiej jest korzystać z [kompozycji komponentow](/docs/components-and-props.html#composing-components).

Reviewer: ```suggestion
>Ta strategia jest szczególnie przydatna, gdy części strony oparte na Reakcie są od siebie odizolowane. W kodzie reactowym łatwiej jest korzystać z [kompozycji komponentów](/docs/components-and-props.html#composing-components).
```

> ### Wskazówka: Minifikuj JavaScript dla środowiska produkcyjnego {#tip-minify-javascript-for-production}

Reviewer 2: Może:

```suggestion
### Wskazówka: Minifikuj JavaScript dla potrzeb środowiska produkcyjnego {#tip-minify-javascript-for-production}
```

> Przed wdrozeniem strony na środowisko produkcyjne, pamiętaj że nie zminifikowany kod JavaScript może znacznie spowolnić działanie Twojej strony.

Reviewer: ```suggestion
Przed wdrożeniem strony na środowisko produkcyjne, pamiętaj, że niezminifikowany kod javascriptowy może znacznie spowolnić działanie twojej strony.
```

> Przed wdrożeniem strony na środowisko produkcyjne, pamiętaj, że niezminifikowany kod javascriptowy może znacznie spowolnić działanie twojej strony.

Reviewer 2: ```suggestion
Przed wdrożeniem strony do środowiska produkcyjnego, pamiętaj, że niezminifikowany kod javascriptowy może znacznie spowolnić działanie twojej strony.
```

> Jeżeli już zminifikowałeś skrypty aplikacji, **Twoja strona internetowa będzie gotowa dla środowiska produkcyjnego** jeżeli wdrożony kod ładuje wersje Reacta zakonczoną `production.min.js`:

Reviewer: ```suggestion
Jeśli twoje skrypty są już zminifikowane, **aplikacja będzie gotowa dla środowiska produkcyjnego**, jeżeli wdrożony kod będzie ładował wersję Reacta zakończoną `production.min.js`:
```

> Jeśli twoje skrypty są już zminifikowane, **aplikacja będzie gotowa dla środowiska produkcyjnego**, jeżeli wdrożony kod będzie ładował wersję Reacta zakończoną `production.min.js`:

Reviewer 2: Moim zdaniem wersja angielska jest tu dosyć pokraczna. Proponuję:

```suggestion
Jeżeli twoje skrypty są już zminifikowane i jeśli wdrożony dokument HTML ładuje wersję Reacta z końcówką `production.min.js` w nazwie, wówczas twoja aplikacja będzie gotowa do użytku w środowisku produkcyjnym:
```

```js
<script src="https://unpkg.com/react@16/umd/react.production.min.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js" crossorigin></script>
```

> Jeżeli nie wykonujesz minifikacji dla swoich skryptow, [oto jeden ze sposobów aby to zrobić](https://gist.github.com/gaearon/42a2ffa41b8319948f9be4076286e1f3).

Reviewer: ```suggestion
Jeżeli nie masz jeszcze skonfigurowanej minifikacji skryotów, [oto jeden ze sposobów, aby to zrobić](https://gist.github.com/gaearon/42a2ffa41b8319948f9be4076286e1f3).
```

## Opcjonalnie: Wypróbuj Reacta z JSX {#optional-try-react-with-jsx}

> W przykładach powyżej opieramy się tylko na funkcjach, które są natywnie obsługiwane przez przeglądarki. Dlatego wywołaliśmy funkcję JavaScript, aby powiedzieć Reactowi, co wyświetlić:

Reviewer: ```suggestion
W przykładach powyżej opieramy się tylko na funkcjonalnościach, które są natywnie obsługiwane przez przeglądarki. Dlatego wywołaliśmy funkcję javascriptową, aby poinformować Reacta, co ma wyświetlić:
```

> '''js

Reviewer: ```suggestion
```js
```

const e = React.createElement;

> // Wyświetl "Like" <button>

Reviewer: ```suggestion
// Wyświetl przycisk "Like"
```

return e(

> 'button',

Reviewer: ```suggestion
  'button',
```

> { onClick: () => this.setState({ liked: true }) },

Reviewer: ```suggestion
  { onClick: () => this.setState({ liked: true }) },
```

> 'Like'

Reviewer: ```suggestion
  'Like'
```

);

> '''

Reviewer: ```suggestion
```
```

> Jednakże React również oferuje możliwość użycia [JSX](/docs/introducing-jsx.html):

Reviewer: ```suggestion
Jednak React oferuje również możliwość użycia składni [JSX](/docs/introducing-jsx.html):
```

> '''js

Reviewer: ```suggestion
```js
```

> // Wyświetl "Like" <button>

Reviewer: ```suggestion
// Wyświetl przycisk "Like"
```

return (

> <button onClick={() => this.setState({ liked: true })}>

Reviewer: ```suggestion
  <button onClick={() => this.setState({ liked: true })}>
```

> Like

Reviewer: ```suggestion
Lubię to!
```

> </button>

Reviewer: ```suggestion
  </button>
```

);

> '''

Reviewer: ```suggestion
```
```

> Te dwa fragmenty kodu są równoważne. Mimo iż **składnia JSX jest [całkowicie opcjonalna](/docs/react-without-jsx.html)**, wiele osób uważa, że jest pomocna przy pisaniu kodu dla UI -- zarówno w Reakcie, jak i z innymi bibliotekami.

Reviewer 2: "równoważne" czy "równoznaczne"? A może "działają w ten sam sposób"

Author: Dodalem:  "działają w ten sam sposób"

> Te dwa fragmenty kodu są równoważne. Ponieważ **JSX jest [całkowicie opcjonalny](/docs/react-without-jsx.html)**, wiele osób uważa, że jest to pomocne przy pisaniu kodu UI -- zarówno z React, jak i z innymi bibliotekami.

Reviewer: ```suggestion
Te dwa fragmenty kodu są równoważne. Mimo iż **składnia JSX jest [całkowicie opcjonalna](/docs/react-without-jsx.html)**, wiele osób uważa, że jest pomocna przy pisaniu kodu dla UI -- zarówno w Reakcie, jak i z innymi bibliotekami.
```

> Możesz wypróbować JSX korzystając z [tego konwertera online](http://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=Q&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2%2Cstage-3&prettier=true&targets=Node-6.12&version=6.26.0&envVersion=).

Reviewer: ```suggestion
Możesz wypróbować składnię JSX korzystając z [tego konwertera online](http://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=Q&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2%2Cstage-3&prettier=true&targets=Node-6.12&version=6.26.0&envVersion=).
```

> ### Szybko wypróbuj JSX {#quickly-try-jsx}

Reviewer: ```suggestion
### Szybki test składni JSX {#quickly-try-jsx}
```

> Najszybszym sposobem na wyprobowanie JSX w Twoim projekcie jest dodanie znacznika '<script>' do strony:

Reviewer: ```suggestion
Najszybszym sposobem na wypróbowanie składni JSX w projekcie jest dodanie następującego znacznika `<script>` do strony:
```

> '''html

Reviewer: ```suggestion
```html
```

<script src="https://unpkg.com/babel-standalone@6/babel.min.js"></script>

> '''

Reviewer: ```suggestion
```
```

> Teraz możesz użyć JSX w dowolnym znaczniku '<script>' dodając do niego atrybut 'type="text/babel"'. Tutaj jest [przykład pliku HTML z JSX](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html), który możesz pobrać i zapoznać się.

Reviewer: ```suggestion
Od teraz możesz używać składni JSX wewnątrz dowolnego znacznika `<script>`, który zaopatrzony jest w atrybut `type="text/babel"`. Tutaj jest [przykład pliku HTML z JSX](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html), który możesz pobrać i wypróbować.
```

> Takie podejście jest dobre do nauki i tworzenia projektów demo. Powoduje to jednak spowolnienie działania strony i **nie jest odpowiednie na środowisko produkcyjne**. Gdy będziesz gotowy, aby przejść dalej, usuń znacznik '<script>' i atrybut 'type="text/babel"' który dodałeś. Zamiast tego, w następnej sekcji skonfigurujesz preprocesor JSX, aby automatycznie konwertować wszystkie znaczniki '<script>'.

Reviewer: ```suggestion
Takie podejście jest dobre do nauki i tworzenia projektów na prezentacje. Powoduje to jednak spowolnienie działania strony i **nie jest odpowiednie dla środowiska produkcyjnego**. Kiedy już zdecydujesz, że możesz iść naprzód z materiałem, usuń ostatnio dodany znacznik `<script>` i atrybuty `type="text/babel"` z innych znaczników. W następnym podrozdziale skonfigurujemy preprocesor składni JSX, który będzie konwertował zawartość wszystkich znaczników `<script>` automatycznie.
```

### Dodaj JSX do projektu {#add-jsx-to-a-project}

> Dodanie JSX do projektu nie wymaga skomplikowanych narzędzi takich jak bundler lub serwer deweloperski. Zasadniczo, dodanie JSX **nie jest jak dodanie preprocesora CSS.** Jedynym wymaganiem jest [Node.js](https://nodejs.org/) zainstalowany na komputerze.

Reviewer: ```suggestion
Dodanie JSX do projektu nie wymaga skomplikowanych narzędzi, takich jak bundler czy serwer deweloperski. Zasadniczo dodawanie JSX **przypomina dodawanie preprocesora CSS**. Jedynym wymaganiem jest posiadanie zainstalowanego na komputerze środowiska [Node.js](https://nodejs.org/).
```

> Przejdź do folderu projektu i wklej dwie komendy:

Reviewer: ```suggestion
Przejdź do folderu projektu i wywołaj dwie komendy:
```

> 1. **Krok 1:** Uruchom 'npm init -y' (jeżeli sie nie uda [tutaj jest opisana poprawka](https://gist.github.com/gaearon/246f6380610e262f8a648e3e51cad40d))

Reviewer: ```suggestion
1. **Krok 1:** Uruchom `npm init -y` (jeżeli wywołanie się nie powiedzie - [tutaj opisano, jak obejść problem](https://gist.github.com/gaearon/246f6380610e262f8a648e3e51cad40d))
```

> 2. **Krok 2:** Uruchom 'npm install babel-cli@6 babel-preset-react-app@3'

Reviewer: ```suggestion
2. **Krok 2:** Uruchom `npm install babel-cli@6 babel-preset-react-app@3`
```

>Wskazówka
>

> >Korzystamy z **npm tylko do instalacji preprocesora JSX;** nie będziesz go potrzebował do niczego innego. Zarówno Rect, jak i kod aplikacji mogą pozostać bez zmian w znaczniku '<script>'

Reviewer: ```suggestion
>**Korzystamy tu z `npm` tylko do instalacji preprocesora JSX**; nie będzie on potrzebny do niczego innego. Zarówno React, jak i kod aplikacji mogą pozostać bez zmian w znacznikach `<script>`.```

> Gratulacje! Właśnie dodałeś do swojego projektu **gotowa konfiguracje JSX**.

Reviewer: ```suggestion
Gratulacje! Właśnie udało ci się dodać do swojego projektu **konfigurację JSX gotową do wdrożenia na środowisko produkcyjne**.
```

> Gratulacje! Właśnie udało ci się dodać do swojego projektu **konfigurację JSX gotową do wdrożenia na środowisko produkcyjne**.

Reviewer 2: ```suggestion
Gratulacje! Właśnie udało ci się dodać do swojego projektu **konfigurację JSX gotową do wdrożenia do środowiska produkcyjnego**.
```

### Uruchom preprocesor JSX {#run-jsx-preprocessor}

> Utwórz folder o nazwie 'src' i uruchom polecenie w terminalu:

Reviewer: ```suggestion
Utwórz folder o nazwie `src` i uruchom polecenie w terminalu:
```

> '''

Reviewer: ```suggestion
```
```

npx babel --watch src --out-dir . --presets react-app/prod

> '''

Reviewer: ```suggestion
```
```

> >Notatka

Reviewer: ```suggestion
>Uwaga
```

>

> >'npx' nie jest literówka -- jest to [narzędzie do uruchamiania pakietów dostępne w npm 5.2+](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b).

Reviewer: ```suggestion
>`npx` to nie literówka -- jest to [narzędzie do uruchamiania pakietów, dostępne w npm 5.2+](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b).
```

>

> >Jeśli zobaczysz komunikat o błędzie "Błędnie zainstalowałeś pakiet 'babel'", mogłeś pominąć [poprzedni krok](#add-jsx-to-a-project). Wykonaj go w tym samym folderze, i spróbuj ponownie.

Reviewer: ```suggestion
>Jeśli wyświetli się komunikat o błędzie: "You have mistakenly installed the `babel` package" (pol. "Błędnie zainstalowano pakiet `babel`"), możliwe, że pominięty został [poprzedni krok](#add-jsx-to-a-project). Wykonaj go w tym samym folderze i spróbuj ponownie.
```

> Nie czekaj na jego zakończenie -- to polecenie uruchamia automatyczne śledzenie dla JSX.

Reviewer: ```suggestion
Nie czekaj na jego zakończenie -- to polecenie uruchamia skrypt automatycznie śledzący zmiany w kodzie ze składnią JSX.
```

> Jeśli utworzysz teraz plik o nazwie 'src/like_button.js' wraz z **[przykładowym kodem JSX](https://cdn.rawgit.com/gaearon/c8e112dc74ac44aac4f673f2c39d19d1/raw/09b951c86c1bf1116af741fa4664511f2f179f0a/like_button.js)**, obserwator utworzy przeprocesowany plik 'like_button.js' z kodem JavaScript odpowiednim dla przeglądarki. Gdy edytujesz plik źródłowy, korzystając z JSX, transformacja zostanie ponownie automatycznie uruchomiona.

Reviewer: ```suggestion
Jeśli utworzysz teraz plik o nazwie `src/like_button.js` przy użyciu **[tego przykładowego kodu JSX](https://cdn.rawgit.com/gaearon/c8e112dc74ac44aac4f673f2c39d19d1/raw/09b951c86c1bf1116af741fa4664511f2f179f0a/like_button.js)**, obserwator (ang. *watcher*) utworzy przetworzony plik o nazwie `like_button.js` zawierający kod javascriptowy zrozumiały dla przeglądarki. Gdy wyedytujesz plik źródłowy z kodem JSX, transformacja zostanie ponownie uruchomiona automatycznie.
```

> Dodatkowo umożliwi to również korzystanie z nowych składni JavaScript, takich jak klasy, bez obawy o brak kompatybilności ze starszymi przeglądarkami. Narzędzie, które właśnie użyliśmy, nazywa się Babel, i możesz dowiedzieć się więcej na jego temat z [dokumentacji](http://babeljs.io/docs/en/babel-cli/).

Reviewer: ```suggestion
Dodatkowo umożliwi to również korzystanie z nowych składni JavaScript, takich jak klasy, bez obawy o brak kompatybilności ze starszymi przeglądarkami. Narzędzie, którego właśnie użyliśmy, nazywa się Babel. Możesz dowiedzieć się więcej na jego temat z [dokumentacji](http://babeljs.io/docs/en/babel-cli/).
```

> Jeśli zauważyłeś, że czujesz się komfortowo z narzędziami do kompilacji i chcesz by robiły dla Ciebie więcej, [następna sekcja](/docs/create-a-new-react-app.html) opisuje niektóre z najbardziej popularnych i przystępnych narzędzi. Jeżeli nie -- znaczniki script będą wystarczające.

Reviewer: ```suggestion
Jeśli uważasz, że czujesz się już komfortowo z narzędziami do kompilacji i chcesz, by robiły za ciebie więcej, zajrzyj do [następnego rozdziału](/docs/create-a-new-react-app.html), w którym opisaliśmy niektóre z najbardziej popularnych i przystępnych narzędzi. Jeżeli nie -- cóż, znaczniki `script` też wystarczą!
```
